### PR TITLE
Prevent crash on empty area filter

### PR DESCRIPTION
**Pull request by bschne**

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs21/controller/UsersApiController.java [hunk 1]
```diff
@@ -188,7 +188,7 @@ public UsersApiController(NativeWebRequest request, UserService userService, Cha
 
     @Override
     public ResponseEntity<List<UserOverviewDto>> getUsers(@ApiParam(value = "") @Valid AreaFilterDto areaFilter,@ApiParam(value = "") @Valid RadiusFilterDto radiusFilter) throws Exception {
-        if(areaFilter != null) {
+        if(areaFilter != null && areaFilter.getVisibleArea().size() > 0) {
```
Comment (KseniaBel) [inline src/main/java/ch/uzh/ifi/hase/soprafs21/controller/UsersApiController.java:191]: This is checked by API. Visible area has minItems: 3.

**Comment by bschne** [comment]
The API client function takes two params, “opts” and the callback.

This crashed on the next line when I passed either {} or undefined for the opts parameter b/c index 0 was out of range, and worked when I put in the change.

Do you have an idea why the validation may not work as intended? Or was this tested and might be an issue with how the client composes the request? 🤔

**Comment by KseniaBel** [comment]
Does the positive flow work? 
I will check this corner case a bit later.

**Comment by bschne** [comment]
Don’t know since I haven’t gotten around to integrating the API on the Map yet. Will let you know once I do.

—
+41 79 379 70 31

**Review by KseniaBel — COMMENTED** [review]
I created additional ticket for this problem The fix will be provided there

**Comment by bschne** [comment]
We are solving this issue with #124 instead.

**Pull request closed without merging**